pinot-common/src/main/java/com/linkedin/pinot/common/metadata/stream/KafkaStreamMetadata.java

```diff
@@ -145,7 +151,7 @@ public boolean equals(Object o) {
   @Override
   public int hashCode() {
     int result = hashCodeOf(_kafkaTopicName);
-    result = hashCodeOf(result, _consumerType);
+    result = hashCodeOf(result, _consumerTypes);
```

Comment:
This doesn't match with the implementation of equals. Two `KafkaStreamMetadata` are equal even if their consumer types are in a different order, but they'll have a different hashCode. In this case, this means that if two instances of the same `KafkaStreamMetadata` with different consumer types order are put in the same set, the behavior won't be deterministic and you'll get either one or two items in the set, depending on whether they fall in the same slot or not.